Bucketeer/Sources/Public/BKTClient.swift

```diff
@@ -70,29 +64,38 @@ public class BKTClient {
             }
         }
     }
+    
+    private func destroy() {
+        taskScheduler?.invalidate()
+        taskScheduler = nil
+        execute { [weak self] in
+            // must destroy in the intenal queue to prevent race condition
+            self?.component.destroy()
+        }
+    }
 }
 
 extension BKTClient {
     public static func initialize(config: BKTConfig, user: BKTUser, timeoutMillis: Int64 = 5000, completion: ((BKTError?) -> Void)? = nil) throws {
         guard (Thread.isMainThread) else {
```

Comment:
I forgot to mention. Because we don't use the applicationState anymore, there is no need to make the SDK initialized on the main thread. Please remove it from destroy, too.

Reply:
changed https://github.com/bucketeer-io/ios-client-sdk/pull/40/commits/c3cbc81e92dff55e903b98379cb980df212bf654